nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/WebIdentityCredentialsStrategy.java

```diff
@@ -0,0 +1,219 @@
+        }
+
+        final StsClientBuilder stsClientBuilder = StsClient.builder().httpClient(httpClientBuilder.build());
+
+        if (stsRegionId != null) {
+            stsClientBuilder.region(Region.of(stsRegionId));
+        }
+
+        if (stsEndpoint != null && !stsEndpoint.isEmpty()) {
+            stsClientBuilder.endpointOverride(URI.create(stsEndpoint));
+        }
+
+        final StsClient stsClient = stsClientBuilder.build();
+
+        return new WebIdentityRefreshingCredentialsProvider(stsClient, tokenProvider, roleArn, roleSessionName, sessionSeconds);
+    }
+
+    @Override
+    public com.amazonaws.auth.AWSCredentialsProvider getCredentialsProvider(final PropertyContext propertyContext) {
+        throw new UnsupportedOperationException("AWS Java SDK v1 is not supported by Web Identity strategy");
+    }
+
+    private static final class WebIdentityRefreshingCredentialsProvider implements AwsCredentialsProvider {
+        private static final Duration SKEW = Duration.ofSeconds(60);
+
+        private final StsClient stsClient;
+        private final OAuth2AccessTokenProvider oauth2AccessTokenProvider;
+        private final String roleArn;
+        private final String roleSessionName;
+        private final Integer sessionSeconds;
+
+        private volatile AwsSessionCredentials cached;
+        private volatile Instant expiration;
+
+        private WebIdentityRefreshingCredentialsProvider(final StsClient stsClient,
+                                                         final OAuth2AccessTokenProvider oauth2AccessTokenProvider,
+                                                         final String roleArn,
+                                                         final String roleSessionName,
+                                                         final Integer sessionSeconds) {
+            this.stsClient = Objects.requireNonNull(stsClient, "stsClient required");
+            this.oauth2AccessTokenProvider = Objects.requireNonNull(oauth2AccessTokenProvider, "OAuth2AccessTokenProvider required");
+            this.roleArn = Objects.requireNonNull(roleArn, "roleArn required");
+            this.roleSessionName = Objects.requireNonNull(roleSessionName, "roleSessionName required");
+            this.sessionSeconds = sessionSeconds;
+        }
+
+        @Override
+        public AwsCredentials resolveCredentials() {
+            final Instant now = Instant.now();
+            final AwsSessionCredentials current = cached;
+            final Instant currentExpiration = expiration;
+            if (current != null && currentExpiration != null && now.isBefore(currentExpiration.minus(SKEW))) {
+                return current;
+            }
+
+            synchronized (this) {
+                if (cached != null && expiration != null && Instant.now().isBefore(expiration.minus(SKEW))) {
+                    return cached;
+                }
+
+                final String webIdentityJwt = getWebIdentityToken();
```

Comment:
Minor, but I recommend retaining `Token` in the variable name:
```suggestion
                final String webIdentityToken = getWebIdentityToken();
```